contrib/ext_editor.lua

```diff
@@ -83,26 +83,29 @@ local PS = dt.configuration.running_os == "windows" and  "\\"  or  "/"
 -- translation
 local gettext = dt.gettext
 gettext.bindtextdomain(MODULE_NAME, dt.configuration.config_dir..PS.."lua"..PS.."locale"..PS)
-
 local function _(msgid)
   return gettext.dgettext(MODULE_NAME, msgid)
 end
 
 
 -- number of valid entries in the list of external programs
-local n_entries = 0
+local n_entries
+
+
+-- allowed file extensions, to exclude RAW, which cannot be edited externally
+local allowed_file_types = {"JPG", "jpg", "JPEG", "jpeg", "TIF", "tif", "TIFF", "tiff", "EXR", "exr"}
 
 
 -- last used editor initialization
 if not dt.preferences.read(MODULE_NAME,"initialized", "bool") then
-	dt.preferences.write(MODULE_NAME,"lastchoice", "integer", 1)
+	dt.preferences.write(MODULE_NAME,"lastchoice", "integer", 0)
 	dt.preferences.write(MODULE_NAME,"initialized", "bool", true)
 	end 
-local lastchoice = 1
+local lastchoice = 0
 
 
 -- update lists of program names and paths, as well as combobox ---------------
-local function UpdateProgramList(combobox, button_pressed) 
+local function UpdateProgramList(combobox, button1, button2, update_button_pressed) 
```

Comment:
button1 and button2 are not the best name, perhaps rename them to button_edit and button_edit_copy